src/main/java/services/DatafeedEventsServiceV1.java

```diff
@@ -0,0 +1,182 @@
+package services;
+
+import clients.SymBotClient;
+import configuration.LoadBalancingMethod;
+import configuration.SymConfig;
+import configuration.SymLoadBalancedConfig;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+import model.DatafeedEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DatafeedEventsServiceV1 extends AbstractDatafeedEventsService {
+
+    private final Logger logger = LoggerFactory.getLogger(DatafeedEventsServiceV1.class);
+
+    private static final int MAX_BACKOFF_TIME = 5 * 60; // five minutes
+
+    private String datafeedId = null;
+    private ExecutorService pool;
+    private AtomicBoolean stop = new AtomicBoolean();
+    private static int THREADPOOL_SIZE;
+    private static int TIMEOUT_NO_OF_SECS;
+
+    public DatafeedEventsServiceV1(SymBotClient client) {
+        super(client);
+
+        int threadPoolSize = client.getConfig().getDatafeedEventsThreadpoolSize();
+        THREADPOOL_SIZE = threadPoolSize > 0 ? threadPoolSize : 5;
```

Comment:
I would be in favor of putting THREADPOOL_SIZE as a non-static field since it is modified in a non-static method.

Reply:
As suggestion from @symphony-thibault , we won't change the legacy code of v1, so I only copy-paste code from legacy code for this class. But your comment is right, I also have no idea why THREADPOOL_SIZE is static here!

Reply:
Agreed, I want to avoid side effects or breaking changes in the V1 implementation.